src/Session.js

```diff
@@ -34,11 +34,11 @@ export default class Session {
         if (this.options.sessionToken && !requireNewToken) {
             return this.options.sessionToken
         }
-        if (this.state === Session.State.LOGGING_IN) {
-            return Promise.reject(new Error('Already logging in!'))
+        if (this.state !== Session.State.LOGGING_IN) {
+            this.state = Session.State.LOGGING_IN
+            this.sessionTokenPromise = this.loginFunction()
         }
-        this.state = Session.State.LOGGING_IN
-        return this.loginFunction().then((tokenObj) => {
+        return this.sessionTokenPromise.then((tokenObj) => {
```

Comment:
I think there's a bug here - now a new `then` handler is added to this single promise for every call to `getSessionToken`! I think you should attach the `then` handler on line 39 where you create the Promise, and on this line just `return this.sessionTokenPromise`.

Please also add a `catch` block that sets `this.state` to `Session.State.LOGGED_OUT` and throws the error onwards. Otherwise if the login attempt throws, the `Session` will be forever stuck to `LOGGING_IN` and subsequent attempts cannot succeed.